subsys/bluetooth/host/adv.c

```diff
@@ -997,6 +997,23 @@ int bt_le_adv_start_ext(struct bt_le_ext_adv *adv,
 
 static void adv_timeout(struct k_work *work);
 
+void bt_le_lim_adv_cancel_timeout(struct bt_le_ext_adv *adv)
+{
+	int rc;
+
+	rc = k_work_delayable_remaining_get(&adv->timeout_work);
```

Comment:
This is not safe, there might be no remaining time i.e timeout is out, but work is still not executed.
I suggest you just cancel always. That's the recommended approach.